### PR TITLE
Added documentation for the new GCP storage scaler

### PR DESCRIPTION
**Pull request by RamCohen**
Added documentation for the new GCP storage scaler

Signed-off-by: Ram Cohen <ram.cohen@gmail.com>

Relates to https://github.com/kedacore/keda/issues/2628

### CODE REVIEW

File: content/docs/2.7/scalers/gcp-storage.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+- `bucketName` - Name of the bucket in GCP Storage.
+- `targetObjectCount` - Average target value to trigger scaling actions. (Default: `100`, Optional)
+- `maxBucketItemsToScan` - When to stop counting how many objects are in the bucket. (Default: `1000`, Optional)
+As counting the number of objects involves iterating over their metadata it is advised to set this number to the value of `targetObjectCount` * `maxReplicaCount`.
+
+You can also optionally assign a name to the metric using the `metricName` value. If not specified, the `metricName` will be generated automatically based on the trigger index and `bucketName`. For example: **s0-gcp-storage-bucketName**. The value will be prefixed with `s{triggerIndex}-gcp-storage-`.
+
+You can provide in the metadata either `credentialsFromEnv` or `credentialsFromEnvFile`.
+- `credentialsFromEnv` - Set to the name of the environment variable that holds the credential information.
+- `credentialsFromEnvFile` - Set to the name of a json file that holds the credential information.
```
Comment (tomkerkhove) [inline content/docs/2.7/scalers/gcp-storage.md:34]: Can you add this to the YAML example above please?

File: content/docs/2.7/scalers/gcp-storage.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+### Authentication Parameters
+You can use `TriggerAuthentication` CRD to configure the authenticate by providing the service account credentials in JSON. 
+
+**Credential based authentication:**
+
+- `GoogleApplicationCredentials` - Service account credentials in JSON.
+
+### Example
```
Comment (tomkerkhove) [inline content/docs/2.7/scalers/gcp-storage.md:43]:
```suggestion
### Examples
#### Using environment variables
```

File: content/docs/2.7/scalers/gcp-storage.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    name: keda-gcp-storage-go
+  triggers:
+  - type: gcp-storage
+    metadata:
+      bucketName: "Transactions"
+      targetObjectCount: "5"
+      credentialsFromEnv: GOOGLE_APPLICATION_CREDENTIALS_JSON
+```
+
+### Example using TriggerAuthentication
```
Comment (tomkerkhove) [inline content/docs/2.7/scalers/gcp-storage.md:62]:
```suggestion
#### Using TriggerAuthentication
```

File: content/docs/2.7/scalers/gcp-storage.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    authenticationRef:
+      name: keda-trigger-auth-gcp-credentials
+    metadata:
+      bucketName: "Transactions"
+      targetObjectCount: "5"
+```
+
+**Identity based authentication:**
+
+You can also use `TriggerAuthentication` CRD to configure the authentication using the associated service account of the running machine in Google Cloud. You only need to create a `TriggerAuthentication` as this example, and reference it in the `ScaledObject`. `ClusterTriggerAuthentication` can also be used if you intend to use it globally in your cluster.
```
Comment (tomkerkhove) [inline content/docs/2.7/scalers/gcp-storage.md:93]: Can you add this before `Example` please? We tend to show all auth options before diving into samples

File: content/docs/2.7/scalers/gcp-storage.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    name: keda-gcp-storage-go
+  triggers:
+  - type: gcp-storage
+    authenticationRef:
+      name: keda-clustertrigger-auth-gcp-credentials
+      kind: ClusterTriggerAuthentication
+    metadata:
+      bucketName: "Transactions"
+      targetObjectCount: "5"
+```
```
Comment (tomkerkhove) [inline content/docs/2.7/scalers/gcp-storage.md:148]: If we cover TriggerAuthentication the nI think we already provide enough help, but can't hurt

**Review by tomkerkhove — APPROVED** [review]
LGTM, just a few alignments - Thank you!

**Review by tomkerkhove — APPROVED** [review]
LGTM - Waiting for feature to be merged

**Comment by tomkerkhove** [comment]
@RamCohen Can you fix the DCO issues please? You can learn more in our [contribution guide](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#i-didnt-sign-my-commit-now-what).

File: content/docs/2.7/scalers/gcp-storage.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+availability = "2.7+"
+maintainer = "Community"
+description = "Scale applications based on the count of objects in a given GCP Storage bucket."
+layout = "scaler"
+go_file = "gcp_storage_scaler"
++++
+
+### Trigger Specification
+
+This specification describes the `gcp-storage` trigger for GCP Storage. It scales based on the count of objects in a given GCS bucket and assumes the worker is responsible for clearing the bucket by deleting/moving the objects once their processing is complete.
```
Comment (arschles) [inline content/docs/2.7/scalers/gcp-storage.md:12]:
```suggestion
This specification describes the `gcp-storage` scaler, which scales Kubernetes workloads based on the count of objects in a given Google Cloud Storage (GCS) bucket. This scaler assumes the worker, when run, will process and clear the bucket by deleting/moving objects therein.
```

File: content/docs/2.7/scalers/gcp-storage.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    bucketName: test-bucket
+    targetObjectCount: '100'
+    maxBucketItemsToScan: '1000'
+    credentialsFromEnv: GOOGLE_APPLICATION_CREDENTIALS_JSON # Optional
+    credentialsFromEnvFile: GOOGLE_APPLICATION_CREDENTIALS_JSON # Optional
+```
+
+**Parameter list:**
+
+- `bucketName` - Name of the bucket in GCP Storage.
```
Comment (arschles) [inline content/docs/2.7/scalers/gcp-storage.md:27]:
```suggestion
- `bucketName` - Name of the bucket in GCS.
```

**Review by arschles — CHANGES_REQUESTED** [review]

File: content/docs/2.7/scalers/gcp-storage.md (hunk 7 above)
Comment (tomkerkhove) [reply content/docs/2.7/scalers/gcp-storage.md:27]: Not everyone knows what GCS stands for though, but one could assume people should know when they use that service.

Still, might be good to not use abbreviations other than GCP for example

**Review by zroubalik — APPROVED** [review]

File: content/docs/2.7/scalers/gcp-storage.md (hunk 6 above)
Comment (RamCohen) [reply content/docs/2.7/scalers/gcp-storage.md]: applied

File: content/docs/2.7/scalers/gcp-storage.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
```
Comment (arschles) [inline content/docs/2.7/scalers/gcp-storage.md:27]: now that you've called it GCS above, I think this and the title (above) should change to GCS as well

```suggestion
- `bucketName` - Name of the bucket in GCS.
```

File: content/docs/2.7/scalers/gcp-storage.md [hunk 9]
```diff
@@ -0,0 +1,122 @@
++++
+title = "Google Cloud Platform Storage"
+availability = "2.7+"
+maintainer = "Community"
+description = "Scale applications based on the count of objects in a given GCP Storage bucket."
```
Comment (arschles) [inline content/docs/2.7/scalers/gcp-storage.md:5]:
```suggestion
description = "Scale applications based on the count of objects in a given Google Cloud Storage (GCS) bucket."
```

**Review by arschles — CHANGES_REQUESTED** [review]

File: content/docs/2.7/scalers/gcp-storage.md (hunk 9 above)
Comment (tomkerkhove) [reply content/docs/2.7/scalers/gcp-storage.md:5]: Yes, I like this more and would use this in all places `Google Cloud Storage (GCS)`

**Review by arschles — APPROVED** [review]

**Pull request merged**